system-contracts/test/L1Messenger.spec.ts

```diff
@@ -0,0 +1,517 @@
+        ethers.utils.hexlify(randomBytes(32)),
+      ]);
+      emulator.logs[1] = secondLogModified;
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(emulator.buildTotalL2ToL1PubdataAndStateDiffs())
+      ).to.be.rejectedWith("reconstructedChainedLogsHash is not equal to chainedLogsHash");
+    });
+
+    it("should revert chainedMessageHash mismatch", async () => {
+      emulator.logs[1] = logs[1];
+      // Buffer.alloc(32, 6), to trigger the revert
+      const wrongMessage = { lengthBytes: logData.currentMessageLengthBytes, content: Buffer.alloc(32, 6) };
+      emulator.messages[0] = wrongMessage;
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(emulator.buildTotalL2ToL1PubdataAndStateDiffs())
+      ).to.be.rejectedWith("reconstructedChainedMessagesHash is not equal to chainedMessagesHash");
+    });
+
+    it("should revert state diff compression version mismatch", async () => {
+      await (
+        await l1Messenger
+          .connect(knownCodeStorageAccount)
+          .requestBytecodeL1Publication(await ethers.utils.hexlify(utils.hashBytecode(bytecodeData.content)), {
+            gasLimit: 130000000,
+          })
+      ).wait();
+      emulator.messages[0] = { lengthBytes: logData.currentMessageLengthBytes, content: logData.message };
+      // modify version to trigger the revert
+      emulator.version = ethers.utils.hexZeroPad(ethers.utils.hexlify(66), 1);
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(emulator.buildTotalL2ToL1PubdataAndStateDiffs())
+      ).to.be.rejectedWith("state diff compression version mismatch");
+    });
+
+    it("should revert extra data", async () => {
+      emulator.version = ethers.utils.hexZeroPad(ethers.utils.hexlify(1), 1);
+      // add extra data to trigger the revert
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(
+            ethers.utils.concat([emulator.buildTotalL2ToL1PubdataAndStateDiffs(), Buffer.alloc(1, 64)])
+          )
+      ).to.be.rejectedWith("Extra data in the totalL2ToL1Pubdata array");
+    });
+  });
+
+  describe("sendL2ToL1Log", async () => {
+    it("should revert when not called by the system contract", async () => {
+      await expect(l1Messenger.sendL2ToL1Log(logData.isService, logData.key, logData.value)).to.be.rejectedWith(
+        "This method require the caller to be system contract"
+      );
+    });
+
+    it("should emit L2ToL1LogSent event when called by the system contract", async () => {
+      await expect(
+        l1Messenger
+          .connect(l1MessengerAccount)
+          .sendL2ToL1Log(logData.isService, ethers.utils.hexlify(logData.key), ethers.utils.hexlify(logData.value))
```

Comment:
if this test checks that everything works fine under `logData.isService = true`, you should either:

- explicitly double check that `logData.isService = true` so that we are sure that this test it any different from the next one
- just provide `true` instead of `logData.isService`

Reply:
fixed